File: app/models/snippet.rb
class Snippet < ApplicationRecord
  belongs_to :user

  validates :title, presence: true, length: { maximum: 32 }
  validates :description, length: { maximum: 128 }
  validates :file_type, presence: true
  before_save do
    file_type.save!
  end

  validates :content, presence: true

end

File: app/javascript/models/Snippet.ts
class Snippet {
  readonly id: string;
  readonly userId: string;
  readonly title: string;
  readonly fileType: string;
  readonly description: string;
  readonly content: object;

  constructor(
    id: string,
    userId: string,
    title: string,
    fileType: string,
    description: string,
    content: object
  ) {
    this.id = id;
    this.userId = userId;
    this.title = title;
    this.fileType = fileType;
    this.description = description;
    this.content = content;
  }
}

export default Snippet;
